docs/applications/big-data/big-data-in-the-linode-cloud-streaming-data-processing-with-apache-storm.md

```diff
@@ -0,0 +1,1095 @@
+
+    -  **zk-image1.conf** - This is the main image configuration file, and the one you'll be modifying the most. Its properties are described in the next step. This file is named `zk-image1.conf` in our example, but if you chose a different image name, yours may vary.
+
+    -  **zoo.cfg** - This is the primary Zookeeper configuration file. See the official [Zookeeper Configuration Parameters documentation](https://zookeeper.apache.org/doc/current/zookeeperAdmin.html#sc_configuration) for details on what parameters can be customized. It's not necessary to enter the cluster's node list in this file. That's done automatically by the script during cluster creation.
+
+    -  **log4j.properties** - This file sets the default logging levels for Zookeeper components. You can also customize these at a node level when a cluster is created.
+
+    -  **zk-supervisord.conf** - The Zookeeper daemon is run under supervision so that if it shuts down unexpectedly, it's automatically restarted by Supervisord. There is nothing much to customize here, but you can refer to the [Supervisord Configuration documentation](http://supervisord.org/configuration.html) if you want to learn more about the options.
+
+3.  Open the image configuration file (in this example, `./zk-image1/zk-image1.conf`) in an editor. Enter or edit values of configuration properties as required. Properties that *must* be entered or changed from their default values are marked as **REQUIRED**:
+
+    -  `DISTRIBUTION_FOR_IMAGE`
+
+        Specify either Ubuntu 14.04 or Debian 8 to use for this image. This guide has *not* been tested on any other versions or distributions.
+
+        All nodes of all clusters created from this image will have this distribution. The default value is `124` corresponding to Ubuntu 14.04 LTS 64-bit. For Debian 8 64-bit, change this value to `140`.
+
+    -  `LABEL_FOR_IMAGE`
+
+        A label to help you differentiate this image from others. This name will be shown if you edit or view your images in the Linode Manager.
+
+    -  `KERNEL_FOR_IMAGE`
+
+        The kernel version provided by Linode to use in this image. The default value is `138`, corresponding to the latest 64-bit kernel provided by Linode. It is recommended that you leave this as the default setting.
+
+    -  `DATACENTER_FOR_IMAGE`
+
+        The Linode datacenter where this image will be created. This can be any Linode datacenter, but cluster creation is faster if the image is created in the same datacenter where the cluster will be created. It's also recommended to create the image in the same datacenter as the Cluster Manager Linode. Select a datacenter that is geographically close to your premises, to reduce network latency.
+
+        This value can either be the datacenter's ID or location or abbreviation. To see a list of all datacenters:
+
+            ./zookeeper-cluster-linode.sh datacenters api_env_linode.conf
```

Comment:
What's the difference between finding data center IDs this way, versus the command used above, `~/storm-linode/linode_api.py datacenters table` ?